eras/shelley/impl/src/Cardano/Ledger/Shelley/TxWits.hs

```diff
@@ -119,27 +120,56 @@ deriving via
     (Era era, NoThunks (Script era)) =>
     (NoThunks (WitnessSetHKD Identity era))
 
-type ShelleyTxWits = WitnessSetHKD Identity
+newtype ShelleyTxWits era = ShelleyTxWitsConstr (WitnessSetHKD Identity era)
+
+deriving newtype instance EraScript era => Eq (ShelleyTxWits era)
+
+deriving newtype instance EraScript era => Show (ShelleyTxWits era)
+
+deriving newtype instance Era era => Generic (ShelleyTxWits era)
+
+instance
+  ( Era era,
+    NFData (Script era),
+    NFData (WitVKey 'Witness (EraCrypto era)),
+    NFData (BootstrapWitness (EraCrypto era))
+  ) =>
+  NFData (ShelleyTxWits era)
+
+instance (Era era, NoThunks (Script era)) => (NoThunks (ShelleyTxWits era))
+
+-- =======================================================
+-- Accessors
+-- =======================================================
+
+lensWitsRaw ::
+  (WitnessSetHKD Identity e -> a) ->
+  (WitnessSetHKD Identity e -> t -> WitnessSetHKD Identity era) ->
+  Lens (ShelleyTxWits e) (ShelleyTxWits era) a t
+lensWitsRaw getter setter =
+  lens
+    (\(ShelleyTxWitsConstr w) -> getter w)
+    (\(ShelleyTxWitsConstr w) val -> ShelleyTxWitsConstr $ setter w val)
 
 -- | Addresses witness setter and getter for `ShelleyTxWits`. The
 -- setter does update memoized binary representation.
 addrShelleyWitsL ::
-  EraTxWits era => Lens' (ShelleyTxWits era) (Set (WitVKey 'Witness (EraCrypto era)))
-addrShelleyWitsL = lens addrWits' (\w s -> w {addrWits = s})
+  Lens' (ShelleyTxWits era) (Set (WitVKey 'Witness (EraCrypto era)))
+addrShelleyWitsL = lensWitsRaw addrWits' (\witsRaw a -> witsRaw {addrWits' = a})
```

Comment:
It might still be in your plans, but just in case you overlooked that part of the ticket since it is a bit subtle. We also need to adjust lens names for consistency `addrShelleyWitsL` -> `addrShelleyTxWitsL` and the same for class functions: `addrWitsL` -> `addrTxWitsL`

This can be easily achieved with a regular search ad replace in all files, so it should be pretty easy. Let me know if you need help with it.

Reply:
Yes I definitely overlooked that, thanks for the reminder. Will add a commit